staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/unstructured/unstructured.go

```diff
@@ -80,6 +80,31 @@ func (obj *UnstructuredList) UnstructuredContent() map[string]interface{} {
 	return obj.Object
 }
 
+func (u Unstructured) DeepCopy() Unstructured {
+	return Unstructured{
+		Object: deepCopyJSON(u.Object).(map[string]interface{}),
+	}
+}
+
+func deepCopyJSON(x interface{}) interface{} {
```

Comment:
This function assumes that we terminate in "simple" leaf node types, right?  Otherwise we'd end in a shallow copy.

I think the assumption holds true for all reasonable uses of unstructured, I just want to be sure I've understood correctly.